lib/utils.js

```diff
@@ -1,67 +1,12 @@
-const url = require('url');
 const crypto = require('crypto');
-const { ipCheck, errors } = require('arsenal');
+const { errors } = require('arsenal');
 
 const { config } = require('./Config');
 const constants = require('../constants');
 
 const utils = {};
 
 /**
- * Get bucket name from the request of a virtually hosted bucket
```

Comment:
I left `isValidBucketName` because it checks if the bucket starts with `constants.mpuBucketPrefix` which I figured would not be relevant for Mystique. I was thinking we might create an equivalent function in the other repo since there may be additional constraints for bucket name validity.

However, since it seems like the rest of the checks are applicable, we could refactor `isValidBucketName` to accept an array of strings to check if `bucketname.startsWith(...)` and move it to Arsenal in case that might be useful. Or we could create a class/function in Arsenal which accepts an array of additional functions as checks to run on `bucketName`. 👍 

`getAllEndpoints` I left because it depends on the local config. I guess I can move this function and pass the endpoints instead of the keys instead.

The rest of the functions I moved were used directly in the routes. I left any functions that were not used by the routes, so I guess most of them are used by the apis and can be moved to `apiUtils` if we want.